src/classes/SubredditConfig.ts

```diff
@@ -15,8 +16,12 @@ import type {Usernote} from '../types/Usernote';
 export class SubredditConfig {
 	private data: RawSubredditConfig;
 
-	constructor (jsonString: string) {
-		this.data = migrateConfigToLatestSchema(JSON.parse(jsonString));
+	constructor (jsonString?: string) {
+		if (jsonString) {
+			this.data = migrateConfigToLatestSchema(JSON.parse(jsonString));
+		} else {
+			this.data = migrateConfigToLatestSchema(DEFAULT_CONFIG);
+		}
```

Comment:
`DEFAULT_CONFIG` always uses the latest schema and will never need to be migrated forward, so the `migrate` call in the second branch is really just masking a type assertion. Let's do the type assertion explicitly instead, and add a TODO to explain why it's necessary despite being not-really-sound.

```suggestion
		if (jsonString) {
			this.data = migrateConfigToLatestSchema(JSON.parse(jsonString));
		} else {
			// TODO: the default config value isn't actually typed correctly, this needs to be cleaned up eventually
			this.data = DEFAULT_CONFIG as RawSubredditConfig;
		}
```